tests/main_test.py

```diff
@@ -0,0 +1,281 @@
+            }]
+        }]
+        self.timeline1 = [{
+            'event': 'created'
+        }, {
+            'event': 'assigned',
+            'assignee': {
+                'login': 'reviewerName1'
+            },
+            'created_at': self._get_past_time(hours=22)
+        }, {
+            'event': 'assigned',
+            'assignee': {
+                'login': 'reviewerName2'
+            },
+            'created_at': self._get_past_time(hours=56)
+        }]
+
+        self.timeline2 = [{
+            'event': 'created'
+        }, {
+            'event': 'assigned',
+            'assignee': {
+                'login': 'reviewerName1'
+            },
+            'created_at': self._get_past_time(hours=23)
+        }, {
+            'event': 'assigned',
+            'assignee': {
+                'login': 'reviewerName2'
+            },
+            'created_at': self._get_past_time(hours=19)
+        }]
+        self.test_template = '{{ username }}\n{{ pr_list }}'
+
+    def mock_all_get_requests(self, mock_request: requests_mock.Mocker) -> None:
+        """Mock all get requests."""
+
+        param_page_1 = '?page=1&per_page=100'
+        param_page_2 = '?page=2&per_page=100'
+        mock_request.get(
+            github_services.PULL_REQUESTS_URL_TEMPLATE.format(
+                self.org_name, self.repo_name) + param_page_1,
+            text=json.dumps(self.pull_response))
+        mock_request.get(
+            github_services.PULL_REQUESTS_URL_TEMPLATE.format(
+                self.org_name, self.repo_name) + param_page_2,
+            text=json.dumps([]))
+
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.org_name, self.repo_name, 123) + param_page_1,
+            text=json.dumps(self.timeline1))
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.org_name, self.repo_name, 123) + param_page_2,
+            text=json.dumps([]))
+
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.org_name, self.repo_name, 234) + param_page_1,
+            text=json.dumps(self.timeline2))
+        mock_request.get(
+            github_services.ISSUE_TIMELINE_URL_TEMPLATE.format(
+                self.org_name, self.repo_name, 234) + param_page_2,
+            text=json.dumps([]))
+
+    def test_executing_main_function_sends_notification(self) -> None:
+        """Test main function to send notification."""
+
+        with requests_mock.Mocker() as mock_request:
+
+            # Mock all get requests.
+            self.mock_all_get_requests(mock_request)
+
+            # Here we are mocking the two post requests, we will use in the below test.
+            # One request for fetching all existing GitHub Discussions data and the next
+            # request to post a comment in the particular GitHub Discussion.
+
+            # Creating a Mock instance.
```

Comment:
This and the next 2 comments aren't necessary. Comments should explain the "why" and "broader picture" behind the code, and not just say literally what the code does. Instead, the comment above should go above these three lines (and it's fine to drop the newline on line 243 so that it's clear the comment refers to this "batch").

Reply:
done.